### DISCUSSION TITLE
The lack of a custom ffmpeg startup limit function when transcoding multiple files causes the operating system to crash fatally.

### DISCUSSION THREAD

**Discussion by lea-nsk** [cwilby/nextcloud-workflow-media-converter#464] [General]
Hello.
Software:
Linux Debian 12;
Nextcloud 29.0.1;
Media conversion + ffmpeg;
Hardware:
CPU: 4 core;
RAM: 4 Gb;
Swap disk part for Linux Debian 12: 8 Gb.
Recently I decided to tidy up my working video archive into a single format for further work with it in Nextcloud. I have about 5000+ video files with different file sizes. The problem at the moment of assigning a folder with many files to be transcoded into another format using Media conversion is that while processing the files, several background tasks are running, pulling off ffmpeg, which causes CPU overload and in the progression of running overflowing RAM with it, itself causing a fatal crash in the operating system. In this situation I don't like the fact that scheduled jobs can't be fully marked or paused. And also Media conversion has no functional controls for enabling/disabling/pausing a scheduled media processing job, including functions for setting limits on RAM usage and CPU load.
![Clipboard01](https://github.com/cwilby/nextcloud-workflow-media-converter/assets/78995154/f5e8b584-224a-451b-a93d-97f80b6c2a39)

In my opinion, Media conversion lacks additional optional settings regarding:
1. Setting a limit on the maximum amount of parallel streaming of files through ffmpeg.
2. Ability to set a manual limit on CPU load usage (e.g. 25, 50, 75, 100%) when using ffmpeg on weak hardware servers.

**Comment by cwilby** [comment]
@lea-nsk Given the way the Nextcloud job engine is built, I agree that these parameters should be added to this project.

I will work on adding those two fields you mentioned, they are good determining factors for whether a job should be retried later.

**Reply by cwilby** [reply]
In addition, note that the ability to limit CPU usage is technically implemented by FFmpeg, but doesn't come with the ability to limit CPU load on a single core.
